merge/update.go

```diff
@@ -250,31 +250,47 @@ func (s *Updater) addBackOwnedItems(merged, pruned *typed.TypedValue, managedFie
 		}
 		managedAtVersion[managerSet.APIVersion()] = managedAtVersion[managerSet.APIVersion()].Union(managerSet.Set())
 	}
-	for version, managed := range managedAtVersion {
-		merged, err = s.Converter.Convert(merged, version)
```

Comment:
Can this be more simply fixed by just not modifying `merged` and starting from the original form of `merged` each iteration? There should really not be path-dependencies sneaking in from the random iteration order of a loop like this.

Reply:
(answering myself: no)